gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -412,45 +413,44 @@ def parse(self, *args, **kwargs) -> bool:
-                    self.rebuild_requested and self.overwrite_allowed
-                ):
-                    if not self.store.has_existing:
-                        self.info(
-                            f"No existing data found. Creating new Zarr at {self.store}."
-                        )
+        ) as cluster:
+            with Client(cluster) if not isinstance(self.store, IPLD) else nullcontext() as client:
+                self.info(f"Dask Dashboard for this parse can be found at {cluster.dashboard_link}")
+                try:
+                    # Attempt to find an existing Zarr, using the appropriate method for the store. If there is existing data and there is no
+                    # rebuild requested, start an update. If there is no existing data, start an initial parse. If rebuild is requested and there is
+                    # no existing data or allow overwrite has been set, write a new Zarr, overwriting (or in the case of IPLD, not using) any existing
+                    # data. If rebuild is requested and there is existing data, but allow overwrite is not set, do not start parsing and issue a warning.
+                    if self.store.has_existing and not self.rebuild_requested:
+                        self.info(f"Updating existing data at {self.store}")
+                        self.update_zarr()
+                    elif not self.store.has_existing or (
+                        self.rebuild_requested and self.overwrite_allowed
+                    ):
+                        if not self.store.has_existing:
+                            self.info(
+                                f"No existing data found. Creating new Zarr at {self.store}."
+                            )
+                        else:
+                            self.info(f"Data at {self.store} will be replaced.")
+                        self.write_initial_zarr()
                     else:
-                        self.info(f"Data at {self.store} will be replaced.")
-                    self.write_initial_zarr()
-                else:
-                    raise RuntimeError(
-                        "There is already a zarr at the specified path and a rebuild is requested, "
-                        "but overwrites are not allowed."
+                        raise RuntimeError(
+                            "There is already a zarr at the specified path and a rebuild is requested, "
+                            "but overwrites are not allowed."
+                        )
+                except KeyboardInterrupt:
+                    self.info(
+                        "CTRL-C Keyboard Interrupt detected, exiting Dask client before script terminates"
                     )
-            except KeyboardInterrupt:
-                self.info(
-                    "CTRL-C Keyboard Interrupt detected, exiting Dask client before script terminates"
-                )
-                client.close()
+                    client.close()
```

Comment:
If we stick with nullcontext, we need to remove this line, since nullcontext has no close method. In general though, I don't think this line does anything, since exiting the Client context forces it to close anyways